src/hotspot/share/opto/type.cpp

```diff
@@ -3046,9 +3046,11 @@ TypeOopPtr::TypeOopPtr(TYPES t, PTR ptr, ciKlass* k, bool xk, ciObject* o, int o
         } else if (klass() == ciEnv::current()->Class_klass() &&
                    _offset >= InstanceMirrorKlass::offset_of_static_fields()) {
```

Comment:
You could turn the assert into the check in the enclosing `if`. IMO it makes the code clearer.

Reply:
In that case it would fall into `Instance fields` code below under `} else {` which I don't want.